core/src/main/java/io/neow3j/protocol/rx/JsonRpc2_0Rx.java

```diff
@@ -69,73 +84,73 @@ private Observable<NeoGetBlock> replayBlocksObservableSync(BigInteger startBlock
         }
     }
 
-    public Observable<NeoGetBlock> catchUpToLatestBlockObservable(BigInteger startBlock, boolean fullTransactionObjects,
-            Observable<NeoGetBlock> onCompleteObservable) {
-        // We use a scheduler to ensure this Observable runs asynchronously for users to be
-        // consistent with the other Observables
+    /**
+     * Creates an observable that emits blocks starting at {@code startBlockNumber} up to the most recent block and
+     * continues emitting according {@code onCaughtUpObservable} after that (e.g., stops if {@code
+     * onCaughtUpObservable} is {@code Observable.empty()}).
+     *
+     * @param startBlockIdx          The block index at which to start catching up.
+     * @param fullTransactionObjects If the full transactions objects should be included in the blocks.
+     * @param onCaughtUpObservable   The observable to use after it caught up.
+     * @return the block observable.
+     */
+    public Observable<NeoGetBlock> catchUpToLatestBlockObservable(BigInteger startBlockIdx,
+            boolean fullTransactionObjects, Observable<NeoGetBlock> onCaughtUpObservable) {
+
         return catchUpToLatestBlockObservableSync(
-                startBlock, fullTransactionObjects, onCompleteObservable)
+                startBlockIdx, fullTransactionObjects, onCaughtUpObservable)
+                // We use a scheduler to run this Observable asynchronously
                 .subscribeOn(scheduler);
     }
 
-    public Observable<NeoGetBlock> catchUpToLatestBlockObservable(BigInteger startBlock,
-            boolean fullTransactionObjects) {
-        return catchUpToLatestBlockObservable(startBlock, fullTransactionObjects, Observable.empty());
-    }
```

Comment:
I removed this without side effects just because I wanted to strip down this class to the bear minimum for better maintainability. Whatever neow3j code used this method now simply uses the corresponding overload with an extra boolean argument at the end. Let me know if you don't agree at all with such cleanups.